src/ui/atoms/toast/Toast.tsx

```diff
@@ -30,48 +31,79 @@ type ToastProps = Readonly<{
    * Event handler called when the open state of the dialog changes.
    */
   readonly onOpenChange?: (isOpened: boolean) => void
+  /**
+   * Prevents the toast from closing automatically.
+   */
+  readonly preventAutoClose?: boolean
 }>
 
+// eslint-disable-next-line max-lines-per-function
 export const Toast: React.FC<ToastProps> = ({
   isOpened,
   autoDuration = 3000,
   severityLevel,
   title,
   description,
-  onOpenChange
+  onOpenChange,
+  preventAutoClose
 }: DeepReadonly<ToastProps>) => {
+  const closeButton = (
+    <ToastPrimitive.Close className="okp4-toast-close">
+      <Icon name="close" size={20} />
+    </ToastPrimitive.Close>
+  )
+
+  const renderTitle = (): string | JSX.Element | undefined =>
```

Comment:
Please remove string return type as this function never returns a string

Reply:
I would tend to make this one as a regular JSX component that way:

```
const RenderTitle: React.FC<void> = () => (
)
```

Reply:
I've refactored the code to make it more understandable :)